photon-client/src/main/scala/com/linkedin/photon/ml/hyperparameter/VectorRescaling.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2018 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.hyperparameter
+
+import breeze.linalg.DenseVector
+
+import com.linkedin.photon.ml.util.DoubleRange
+
+/**
+ * Functions to transform/scale forward/backward hyper-parameters in vectors
+ */
+object VectorRescaling {
+
+  /**
+   * Apply forward transformation to a subset of elements in a vector.
+   *
+   * @param vector A DenseVector.
+   * @param transformMap A Map with key-value pairs of indices and names of transform functions.
+   * @return The transformed vector.
+   */
+  def transformForward(
+      vector: DenseVector[Double],
+      transformMap: Map[Int, String]): DenseVector[Double] = {
+
+    val vectorTransformed = vector.copy
+
+    transformMap.foreach { case (index, transform) =>
+      transform match {
+        case "LOG" => vectorTransformed(index) = Math.log10(vectorTransformed(index))
+        case "SQRT" => vectorTransformed(index) = Math.sqrt(vectorTransformed(index))
```

Comment:
Let's extract `"LOG"` and `"SQRT"` out to constants.

Also, we need a catch-all case that throws a user-friendly error message in case an unsupported transformation is passed in, e.g.:

```scala
    case other => throw IllegalArgumentException(s"Unknown transformation: $other")
```